io-engine/src/grpc/v0/mayastor_grpc.rs

```diff
@@ -132,6 +133,46 @@ impl MayastorSvc {
             rw_lock: tokio::sync::RwLock::new(None),
         }
     }
+
+    async fn serialized<T, F>(
+        &self,
+        ctx: GrpcClientContext,
+        nexus_uuid: String,
+        global_operation: bool,
+        f: F,
+    ) -> Result<T, Status>
+    where
+        T: Send + 'static,
+        F: core::future::Future<Output = Result<T, Status>> + Send + 'static,
+    {
+        let lock_manager = ResourceLockManager::get_instance();
+        let fut = AssertUnwindSafe(f).catch_unwind();
+
+        // Grab global operation lock, if requested.
+        let _global_guard = if global_operation {
+            Some(lock_manager.lock().await)
+        } else {
+            None
+        };
+
+        // Grab per-object lock before executing the future.
+        let _resource_guard = lock_manager
+            .get_subsystem(ProtectedSubsystems::NEXUS)
+            .lock_resource(nexus_uuid)
+            .await;
+        let r = fut.await;
+
+        match r {
+            Ok(r) => r,
+            Err(_e) => {
+                warn!("{}: gRPC method panicked, args: {}", ctx.id, ctx.args);
```

Comment:
hmm I wonder if we run fut as a non-cancellable future, or perhaps disable grpc timeout. Otherwise if we have a timeout we'll drop the lock again and open ourselves up to problem we're trying to solve because the mayastor "reactor" futures are not cancellable.